src/current/_includes/releases/v23.1/v23.1.21.md

```diff
@@ -0,0 +1,61 @@
+## v23.1.21
+
+Release Date: May 7, 2024
+
+{% include releases/new-release-downloads-docker-image.md release=include.release %}
+
+<h3 id="v23-1-21-sql-language-changes">SQL language changes</h3>
+
+- The `FORCE_INVERTED_INDEX` hint causes the [optimizer]({% link v23.1/cost-based-optimizer.md %}) to prefer a query plan scan over any [inverted index]({% link v23.1/inverted-indexes.md %}) of the hinted table. An error is emitted if no such query plan can be generated. [#122301][#122301]
+- Introduced three new [cluster settings]({% link v24.1/cluster-settings.md %}) for controlling [table statistics]({% link v23.1/cost-based-optimizer.md %}#table-statistics) forecasting:
+    - [`sql.stats.forecasts.min_observations`]({% link v23.1/cluster-settings.md %}#setting-sql-stats-forecasts-min-observations) is the minimum number of observed statistics required to produce a forecast.
+    - [`sql.stats.forecasts.min_goodness_of_fit`]({% link v23.1/cluster-settings.md %}#setting-sql-stats-forecasts-min-goodness-of-fit) is the minimum R² (goodness of fit) measurement required from all predictive models to use a forecast.
+    - [`sql.stats.forecasts.max_decrease`]({% link v23.1/cluster-settings.md %}#setting-sql-stats-forecasts-max-decrease) is the most a prediction can decrease, expressed as the minimum ratio of the prediction to the lowest prior observation. [#122990][#122990]
+- Added a [session variable]({% link v23.1/set-vars.md %}) `optimizer_use_improved_multi_column_selectivity_estimate`, which if enabled, causes the [optimizer]({% link v23.1/cost-based-optimizer.md %}) to use an improved selectivity estimate for multi-column predicates. This setting will default to `true` on versions 24.2+, and `false` on prior versions. [#123068][#123068]
+
+<h3 id="v23-1-21-operational-changes">Operational changes</h3>
+
+- A minimum [Raft scheduler]({% link v23.1/architecture/replication-layer.md %}#raft) concurrency is now enforced per store so that nodes with many stores do not spread workers too thin. This helps to avoid high scheduler latency across replicas on a store when load is imbalanced. [#120797][#120797]
+
+<h3 id="v23-1-21-bug-fixes">Bug fixes</h3>
+
+- Fixed a bug introduced in v22.2.9 that could cause a slow memory leak that can accumulate when opening many new connections. [#121056][#121056]
+- [Sequence]({% link v23.1/create-sequence.md %}) options for `NO MINVALUE` and `NO MAXVALUE` now match PostgreSQL behavior. Sequence `MINVALUE` and `MAXVALUE` automatically adjust to new types bounds mirroring behavior of PostgreSQL. [#121307][#121307]
```

Comment:
It's not clear what `NO MINVALUE` and `NO MAXVALUE` do. I'm also unclear what "new types bounds" means.

```suggestion
- [Sequence]({% link v23.1/create-sequence.md %}) options for `NO MINVALUE` and `NO MAXVALUE` now match PostgreSQL behavior. Sequence `MINVALUE` and `MAXVALUE` automatically adjust to new types bounds, matching PostgreSQL behavior. [#121307][#121307]
```

Reply:
updated to clarify "new types bounds". I looked at the [tests in the code](https://github.com/cockroachdb/cockroach/pull/121307/files#diff-9c15b618f9f47149f21bc53c432b9357a273c2983485bbdfc66d98def43d7a34R353) and manually ran the tests on v23.1.20:
```
florence@defaultdb> CREATE SEQUENCE seq_change_type_bounds MINVALUE -9223372036854775808 MAXVALUE 9223372036854775807 START 0;
CREATE SEQUENCE

Time: 37ms total (execution 14ms / network 23ms)

florence@defaultdb> SHOW CREATE seq_change_type_bounds;
        table_name       |                                                       create_statement
-------------------------+-------------------------------------------------------------------------------------------------------------------------------
  seq_change_type_bounds | CREATE SEQUENCE public.seq_change_type_bounds MINVALUE -9223372036854775808 MAXVALUE 9223372036854775807 INCREMENT 1 START 0
(1 row)

Time: 58ms total (execution 34ms / network 24ms)

florence@florence-aws-east2-mwb.aws-us-east-2.cockroachlabs.cloud:26257/defaultdb> ALTER SEQUENCE seq_change_type_bounds AS smallint;
ERROR: MINVALUE (-9223372036854775808) must be greater than (-32768) for type INT2
```

In v24.1.0-beta-3 which also has this fix, the `ALTER SEQUENCE` does not get the error. It runs successfully and automatically adjusts the bounds of the new integer datatype, in this case, `smallint`.